### ISSUE TITLE
Different scale for skeletal meshes (armatures) and static meshes

### ISSUE THREAD

**Issue by joric** [poly-hammer/BlenderTools#100]
Basically every export that has both static meshes and skeletal (parented to armature) meshes ends up with different scale for meshes. Skeletal meshes are always 100x smaller. I guess it's somewhere around SCALE_FACTOR = 100 and armature handling code. Inconsistent scaling is inconvenient if you have many meshes in a single file and also affects physical models and vehicles.

How to reproduce:

1) Create 2 default cubes (2x2x2m), drag them to Export collection.
2) Push assets to UE (do not change any settings). Both cubes are 200x200x200 cm.
3) Create empty armature, drag it to collection, parent any cube to the armature.
4) Push assets to UE, skeletal mesh cube is 2x2x2, static mesh cube is 200x200x200.

Blender 4.2, Blender Tools 2.5.

**Upd.** It's not an easy fix, e.g. deleting this line fixes the cubes above but breaks nested hierarchy scale (e.g. wheelcaps).

```diff
         # Todo add to FBX addon
         if ob_obj.type == 'ARMATURE':
-            scale = Vector((scale[0] / SCALE_FACTOR, scale[1] / SCALE_FACTOR, scale[2] / SCALE_FACTOR))
             if bpy.context.scene.send2ue.use_object_origin:
                 loc = Vector((0, 0, 0))
```

A better file to test the hierarchy/armatures/size:

* [gnomecar.blend.zip](https://github.com/user-attachments/files/17727696/gnomecar.blend.zip) (Proper size should be approx 400x400x200 units.)

The easiest way to handle that without fixing the plugin code is to parent armatures to "empty" with scale 100, and check the "Apply Transform" option. I also use "Extensions - Combine assets - Child meshes" plugin setting to get wheels as bones for the car simulation.

Note the generated convex hull on the second picture is 100x smaller than the mesh. This also affects ragdolls because bones are not scaled properly. "Apply Transform" fixes convex hulls and other physics (except nested meshes).

With "Apply Transform" | Without "Apply Transform" 
---|---
![image](https://github.com/user-attachments/assets/adc55deb-da36-4695-b8b3-67570bee6fc1)|![image](https://github.com/user-attachments/assets/303f576f-a3d5-46ea-9440-c294ae6e684e)

The only way to reliably keep hierarchy and get proper size is to rescale all skeletal meshes 100x in Blender before exporting. Also I had to rotate the mesh in Blender because it faces the wrong way, changing export axes to X Forward, Z Up doesn't work too. I also had to merge wheels and wheelcaps meshes because wheelcaps shrink (hierarchical scale doesn't seem to work).

**Comment by DanMcLaughlin** [comment]
Yeah I’ve noticed that too, but haven’t dug into it and pinned it on Send2UE. I separately apply transforms to the mesh, and there’s a different option to apply skeleton transforms. 

I use CGDives Game Rig Tools and have halfway suspected that’s the culprit too btw.

**Comment by JoshQuake** [comment]
Hey guys apologies for delay in response, I've been stupid sick the past week. I will take a look at what's going on on Monday.

**Comment by MalikuMane** [comment]
This seems to be an issue with the import in UE 5.5. I can still export to 5.4 just fine. It breaks if i reimport in 5.5 without send2ue as well.

**Comment by joric** [comment]
I don't have 5.4. Yea I've noticed 5.5 is a little different, Also there's an issue with bone naming. In 5.4 I imported bone names with spaces and it replaced them with dashes, now 5.5 replaces spaces with underscores (e.g. `Bip01-Head` is `Bip01_Head` now) and Mixamo animations made from the old model no longer work.

**Comment by MalikuMane** [comment]
So I disabled the Interchange Framework plugin and it seems to have fixed that scale issue. I diddn't check the issues with the bone naming though.

**Comment by joric** [comment]
That's a lot of dependencies, it also disables Fab and a few other plugins. It does NOT fix the initial 2 cubes issue for me. But general animation import is back to normal as in 5.4 (except bone naming, still underscores). With the Interchange Framework plugin enabled it doesn't even add any animations and collapses the skeleton.

**Comment by MalikuMane** [comment]
Yeah for now it looks like the best option is stick to UE 5.4 or do imports in a seperate 5.4 project and migrate them until there is an actual fix.

**Comment by JoshQuake** [comment]
To update, 5.5.2 did not fix support for import options. Bug report has been sent.

**Comment by JoshQuake** [comment]
closing out as the flag is good enough for now. When interchange is more fleshed out I will make the changes to properly utilize interchange instead of relying on legacy.

**Comment by joric** [comment]
Sorry but the issue is that skeletal meshes are always 100x smaller. I never intended this bug to be anything else.

**Comment by jack-yao91** [comment]
@joric this should fix this. The SceneInterchange FBX importer will make the skeletal meshes scale like you reported here. So should be directly related to your issue, unless there is something more going on. Let us know if this doesn't fix this after updating your unreal project Engine.ini to use the legacy FBX importer.

**Comment by joric** [comment]
It does not fix the issue as per 

> **Disabling the "Interchange Framework" plugin does NOT fix the initial 2 cubes issue for me**.

**Comment by JoshQuake** [comment]
In my testing everything works as intended, so something has to be wrong in your setup. Please send your blend.

**Comment by jack-yao91** [comment]
@joric  You shouldn't need to disable scene interchange, just which FBX importer is the default in your unreal project.

This would be in DefaultEngine.ini
Like in the quick start 
https://poly-hammer.github.io/BlenderTools/send2ue/introduction/quickstart/

**Comment by jack-yao91** [comment]
Otherwise, I would check your blender file and make sure your transformations are applied
